### ISSUE TITLE
MouseEvent causes issues with dispatchEvent  

### ISSUE THREAD

**Issue by charlesbeckwith** [paperjs/paper.js#1652]
After injecting paper scope onto window, MouseEvent is modified and will  cause an error when calling dispatchEvent with a MouseEvent.

This has only been an issue for me since upgrading to 0.12.0.

As a workaround for the time being I simply am creating a reference to the original MouseEvent class and using that when needed. Hopefully this will save some people trouble as it broke some other libraries (FileSaver.js in my case) that worked fine with the previous version of paper.js I was on. 

# Steps

1. Create `new MouseEvent('click')`
2. Obtain reference to dom element
3. Call `dispatchEvent` function on element with the click event created in step 1
4. Inject paper into window `paper.install(window)`
5. Create new MouseEvent as done in step 1
6. Repeat step 3 with MouseEvent from step 5
7. Observe error that MouseEvent object is not an event

# Link to reproduction test-case

https://jsfiddle.net/p9hekrt6/

# Expected result

Click event should be dispatched as expected.

# Additional information
paper.js : 0.12.0
macOS: 10.14.3
chrome: 73.0.3683.103

**Comment by sasensi** [comment]
Yes, there is a known conflict between `Paper.js` internal classes and browser native classes.
We had the same trouble in our testing context and we use the same workaround as you do.
https://github.com/paperjs/paper.js/blob/91a7448c07bc72ad58e1b3fa9407566f2fd2a0f4/test/helpers.js#L41-L44

What is strange though is that you say that it worked before `v0.12.0` and I don't see what could have changed this behavior because `Event` and `MouseEvent` are quite old.
I tried with all the available versions in sketch.paperjs.org but couldn't make it work (see the [sketch](http://sketch.paperjs.org/#V/0.12.1/S/JYy7DoMwDAB/pfISIlV8AMwdmRihg+tYahSwaR6gquq/k6rjSXf3AcGVoYMxcKYnXIHU/dgplZUlt6/C8T3ywpQ1NoZQdkzGts6nDWty26vVCB+XQUviPxpaPAVjbT/HWer1ERnDpl5ygm66f08=)).

Unfortunately, the only solution that I see is to change classes names which would be a breaking change.
@lehni, maybe we can plan that for the next major version ?

Another possible workaround is to use `Paper.js` the scoped way, in JavaScript context and without calling `paper.install(window)`, that will prevent you from this kind of collision.

**Comment by lehni** [comment]
@sasensi the next major version should simply stop injecting into the global scope, I think. It's a terrible practise that has caused issues more than once already...

**Comment by sasensi** [comment]
@lehni, yes, good idea !

**Comment by lehni** [comment]
Luckily it's not doing so by default, but has the user do it by themselves. This kind of saves us :)
